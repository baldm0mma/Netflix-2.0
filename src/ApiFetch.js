import { getAllMovies } from './actions';

const getData = async () => {
  const url = '';
  try {
    const response = await fetch(url);
    const parsed = await response.json();
    getAllMovies(parsed.results);
  } catch (error) {
    getAllMovies(error.message);
    // throw Error(error.message);
  }
};

export default getData;