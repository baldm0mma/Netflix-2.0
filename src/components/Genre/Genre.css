.movies-container {
    display: grid;  
    grid-gap: 20px;  
    grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
    grid-auto-rows: 400px;
    margin:20px;
}

.movies-container > .movie-card > img {
  width: 90%;
  height: 300px;
  margin: 0 auto;
  border-radius:8px;
  object-fit: cover;
}
.movies-container > .movie-card {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  background-color: #5a5a5a;
  box-shadow: 10px 10px 5px 0px rgba(0,0,0,0.75);
}

.movies-container > .movie-card > h3 {
  text-align:center;
  margin:0;
  color:white;
  font-size: .8em;
  line-break: none;
}