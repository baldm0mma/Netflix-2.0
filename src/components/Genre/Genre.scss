.movies-container{
    display: grid;  
    grid-gap: 20px;  
    grid-template-columns: repeat(auto-fit, minmax(200px, 1fr));
    grid-auto-rows: 320px;
    margin:20px;
}
