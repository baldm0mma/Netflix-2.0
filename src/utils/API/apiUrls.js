// import apiKey from '../../apikey';
//import default movie url
export const userSignInURL = 'http://localhost:3001/api/users/';
export const newUserUrl = 'http://localhost:3001/api/users/new';

export const actionUrl = 'https://api.themoviedb.org/3/discover/movie?api_key=6e6bbfd75ee71e1437638b9f37da8fd5&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=1&with_genres=28'

//make this url dynamic with genre ids