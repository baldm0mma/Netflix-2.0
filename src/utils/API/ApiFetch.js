import { newUserUrl, userSignInURL } from './apiUrls';
import apiKey from '../../apikey';
import { cleanMovies } from '../cleanerFunction';
// import { cleanMovies } from '../cleanMovies';

export const findGenres = async () => {
  const genreUrl = `https://api.themoviedb.org/3/genre/movie/list?api_key=${apiKey}&language=en-US`;
  try {
    const response = await fetch(genreUrl);
    const parsed = await response.json();
    return parsed;
  } catch (error) {
    throw Error(error.message);
  }
};

export const fetchSingleGenre = async genreID => {
  const genreUrl = `https://api.themoviedb.org/3/discover/movie?api_key=${apiKey}&language=en-US&sort_by=popularity.desc&include_adult=false&include_video=false&page=1&with_genres=${genreID}`;
  try {
    const response = await fetch(genreUrl);
    const parsed = await response.json();
    const cleanedMovies = await cleanMovies(genreID, parsed.results);
    return cleanedMovies;
  } catch (error) {
    throw Error(error.message);
  }
};
export const searchForMovie = async searchTerm => {
  const genreUrl = `https://api.themoviedb.org/3/search/movie?api_key=${apiKey}&language=en-US&query=${searchTerm}&page=1&include_adult=false`;
  try {
    const response = await fetch(genreUrl);
    const parsed = await response.json();
    const cleaned = cleanMovies(searchTerm, parsed.results);
    return cleaned;
  } catch (error) {
    throw Error(error.message);
  }
};

export const fetchSingleMovie = async movie_id => {
  const singleMovieUrl = `https://api.themoviedb.org/3/movie/${movie_id}?api_key=${apiKey}&language=en-US`;
  try {
    const response = await fetch(singleMovieUrl);
    const parsed = await response.json();
    return parsed;
  } catch (error) {
    throw Error(error.message);
  }
};

const sendUserLogin = async (email, password) => {
  const user = {
    email,
    password
  };
  try {
    const options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(user)
    };
    const response = await fetch(userSignInURL, options);
    const parsed = await response.json();
    return parsed;
  } catch (error) {
    throw Error(
      'Failed to log in; either your email or your password are incorrect. Please try again, or create a new account.',
      error
    );
  }
};

export const fetchUserFavorites = async id => {
  try {
    const response = await fetch(`${userSignInURL}${id}/favorites`);
    const parsed = await response.json();
    await console.log('favorites', parsed);
    return parsed;
  } catch (error) {
    throw Error('Cannot retrieve favorites at this time.', error);
  }
};

const sendNewAccount = async newAccount => {
  try {
    const options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(newAccount)
    };
    const response = await fetch(newUserUrl, options);
    return response;
  } catch (error) {
    throw Error(
      'Sorry, we failed to create your account. Your email or password may already be in use. Please use a different email or password, or attempt to login, as you may already have an account.',
      error
    );
  }
};

const sendFavorite = async favoriteMovie => {
  try {
    const options = {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json'
      },
      body: JSON.stringify(favoriteMovie)
    };
    const response = await fetch(`${userSignInURL}favorites/new`, options);
    const parsed = await response.json();
    return parsed;
  } catch (error) {
    throw Error('Failed to favorite', error);
  }
};

export const deleteFavorite = async (user_id, movie_id) => {
  try {
    const options = {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json'
      }
    };
    const response = await fetch(`${userSignInURL}${user_id}/favorites/${movie_id}`, options);
    const parsed = await response.json();
    return parsed;
  } catch (error) {
    throw Error('Failed to delete favorite', error);
  }
};

export { sendNewAccount, sendUserLogin, sendFavorite };
