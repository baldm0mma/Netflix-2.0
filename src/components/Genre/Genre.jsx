import React, { Component } from 'react'
import { fetchSingleGenre } from '../../utils/API/ApiFetch'
import Card from '../Card/Card'
class Genre extends Component {
  constructor() {
    super()
    this.state = {
      genreMovies: []
    }
  }
  async componentDidMount() {
    const genreMovies = await fetchSingleGenre(this.props.genre.id)
    this.setState({ genreMovies })
  }

  populateMovies() {
    return this.state.genreMovies.map(movie => {
      return <Card movieInfo={movie} />
    })
  }


  render() {
    return (
      <div>
        {this.populateMovies()}
      </div>
    )
  }
}

export default Genre
