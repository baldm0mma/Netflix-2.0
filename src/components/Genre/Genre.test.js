import Genre from './Genre';
import { shallow } from 'enzyme';
import React from 'react';
import { findGenres } from '../../utils/API/ApiFetch';

jest.mock('../../utils/API/ApiFetch');

describe('Genre', () => {
	let wrapper, instance;
	beforeEach(() => {
		wrapper = shallow(<Genre />);
		instance = wrapper.instance();
	});

	describe('component', () => {
		it('should match snapshot', () => {
			expect(wrapper).toMatchSnapshot();
		});
	});

	describe('CDM', () => {
		it('should invoke findGenres', async () => {
			await instance.componentDidMount();
			expect(findGenres).toHaveBeenCalled();
		});

		it('should set state to genres', async () => {
			await instance.componentDidMount();
			expect(wrapper.state('genres')).toEqual([ 'romance' ]);
		});
	});
});
