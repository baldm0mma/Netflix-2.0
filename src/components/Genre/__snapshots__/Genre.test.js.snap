// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Genre component should match snapshot 1`] = `ShallowWrapper {}`;
